superset/dashboards/api.py

```diff
@@ -879,16 +880,21 @@ def thumbnail(self, pk: int, digest: str, **kwargs: Any) -> WerkzeugResponse:
             500:
               $ref: '#/components/responses/500'
         """
-        dashboard = self.datamodel.get(pk, self._base_filters)
+        dashboard = cast(Dashboard, self.datamodel.get(pk, self._base_filters))
```

Comment:
here, note that the `self._base_filters` provides our security against users being able to view each others thumbnails/screenshots

Reply:
This is a VERY good point that I hadn't thought of! Thanks for catching this!